### PR TITLE
NameNsDescription Updates

### PR DESCRIPTION
**Pull request by richard-cox**
UX Changes
- Cancel cross should be a standard colour rather than red
  image
  vs
  image
- Align cancel `x` with select container drop down's arrow (not shown above)
- When choosing to create a namespace set the focus to newly exposed input field (avoids user having to click again)
- Align additional fields for deployment types with fields above (previously fields below were much bigger than above)
- Align margin bottom of NameNsDescription and workload specific field row
  - Also ensure row (with margin) only shows when it has to

UX Questions
- The container selector is now dangling on the end of a row of input fields. I thought to move it to the start, however then the selector is in the middle (reading left to right over rows) of input fields. Not sure where the selector should ultimately end up

Function Fixes
- Ensure additional fields for deployment types show (replica sets was missing for deployments
- Ensure CruResource `namespaceKey` works for paths (they'll be used for the same purpose)
  - align CruResource `namespaceKey` with NameNsDescription `namespaceKey`
  - reduce complexity
  - This follows on from the conversation in https://github.com/rancher/dashboard/pull/6084
 

Overall Visible Changes

Before
image

After
image

### CODE REVIEW

**Comment by catherineluse** [comment]
The reason I made the X red is because I was following the design. Did you let Kenneth know you changed his design?

File: shell/edit/workload/index.vue [hunk 1]
```diff
@@ -901,8 +901,8 @@ export default {
         :mode="mode"
         @change="name=value.metadata.name"
       />
-      <div class="row mb-10">
-        <div v-if="isCronJob" class="col span-4">
+      <div v-if="isCronJob || isReplicable || (isReplicable && isStatefulSet) || containerOptions.length > 1" class="row mb-20">
```
Comment (catherineluse) [inline shell/edit/workload/index.vue:904]: Why are we checking if the workload is both replicable and a StatefulSet? A StatefulSet would always be replicable.

File: shell/edit/workload/index.vue [hunk 2]
```diff
@@ -922,7 +922,7 @@ export default {
             :label="t('workload.replicas')"
           />
         </div>
-        <div v-if="isStatefulSet" class="col span-4">
+        <div v-if="isReplicable && isStatefulSet" class="col span-3">
```
Comment (catherineluse) [inline shell/edit/workload/index.vue:925]: By definition, a StatefulSet is always replicable.

**Review by catherineluse — CHANGES_REQUESTED** [review]

File: shell/edit/workload/index.vue (hunk 1 above)
Comment (richard-cox) [reply shell/edit/workload/index.vue]: This matches the field below

File: shell/edit/workload/index.vue (hunk 2 above)
Comment (richard-cox) [reply shell/edit/workload/index.vue]: I made a tweak, pasting the commit description

- This previously matched the original check for both isReplicable && isStatefulSet
- These work off of the `type` field which does not overlap between types
- It looks like the field `spec.serviceName` is only application to type StatefulSets
  - see https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#statefulset-v1-apps

So now the field is only shown for the correct workload type

**Review by catherineluse — APPROVED** [review]

**Pull request merged**